server/audit_math/raire_utils.py

```diff
@@ -0,0 +1,559 @@
+from __future__ import annotations
+from typing import Type, Callable, Dict, List, Any, Optional, Literal
+import numpy as np
+
+from .sampler_contest import CVR, Contest
+
+
+def ranking(cand: str, ballot: Dict[str, int]) -> int:
+    """
+    Input:
+        cand : string  -   identifier for candidate
+        ballot :       -   mapping between candidate name and their
+                           position in the ranking for a relevant contest
+                           on a given ballot.
+
+    Output:
+        Returns the position of candidate 'cand' in the ranking of the
+        given ballot 'ballot'. Returns -1 if 'cand' is not preferenced on the
+        ballot.
+    """
+    if not cand in ballot:
+        return -1
+
+    rank = ballot[cand]
+    return -1 if not rank else rank
```

Comment:
Now that this function has been simplified, it seems kind of superfluous. If `0` represents a candidate not ranked, then this can just be: `ballot.get(cand, 0)`. I don't see a strong reason to use `-1` as a secondary representation of what `0` already represents. Thoughts?

Reply:
you're not wrong, I suspect there's a degree of cpp -> python getting lost in translation. I'll see if I can fix it.